### ISSUE TITLE
Jobs stuck processing with readPreference=secondayPreferred 

### ISSUE THREAD

**Issue by mack0196** [Hangfire-Mongo/Hangfire.Mongo#355]
Thank you for providing and maintaining Hangfire.Mongo. We use Hangfire in applications used in house using Hangfire.Mongo storage. 

While trying to optimize MongoDB cluster performance, we changed MongoDB connection string to readPreference = secondaryPreferred. After those changes we witnessed jobs staying in a processing state and the enqueue counts growing but viewing the queues it shows they are empty.

Are these known issues using MongoDB connection string with readPreference secondaryPreferred?

**Comment by gottscj** [comment]
@mack0196,

Could elaborate a bit more on your setup?
Are you using atlas?
which `CheckQueuedJobsStrategy` do you use?

Thanks!

**Comment by mack0196** [comment]
We have the default, so the processing jobs aren't removed automatically.

We found that switching read preference in mongo connection strings to secondaryPreferred resulted in jobs that stayed in a processing state. We investigated and couldn't find evidence that the business of the jobs was failing (i.e., no exceptions in logs). After testing with readPreference primaryPreferred this symptom went away. The symptoms also correlate with times our Mongo cluster was under high load.
